Mage.Server/src/main/xml-resources/jaxb/Config/Config.xsd

```diff
@@ -17,6 +17,18 @@
 
     <xs:element name="server">
         <xs:complexType>
+            <xs:sequence>
```

Comment:
Can you another config structure?
* `multihomes="true/false"` in `<server>`
* `<home>` in `<multihomes>`

![shot_201226_154816](https://user-images.githubusercontent.com/8344157/103150869-cbcd6e00-4791-11eb-8932-bb14a8532c6f.png)

Reply:
The `multihome` boolean parameter is already defined in the `xsd` (see \[[1](https://github.com/magefree/mage/pull/7298/files#diff-9373f84b36748c0e76d908e452fed2217b7a1b792ff46605d6fc924a7d78f3e9R31)]).

I'd rather keep the home elements in the server element as semantically it is more meaningful to have the structure contain all the valid and invalid combination instead of dispersing the options.

Reply:
That's incorrect. You can't occupy full `<server>` element by `<home>` tags. So `<multihomes>` element is necessary anyway.

Reply:
@JayDi85: I'm not sure I follow what you are saying.

> You can't occupy full `<server>` element by `<home>` tags

Do you mean that you think that it is not a good solution? Based on the fact that it is possible to just add child nodes to the server element without any issue, clearly it's not a technical problem that you are referring to. In fact, there are multiple tests that are able to parse the file with and without the child home elements.

Assuming you are concerned about how the information is organised, I still believe that the addresses should be saved in proximity to the server element because the other information (for the standard connection parameters) are in that node. Alternatively to adding the homes in the server element, I could create a child node in the server node as follows:

```xml
<server ...>
  <multihomes>
    <home ... />
    <home ... />
  </multihomes>
</server>
```

Alternatively, I could rename the elements to be less connected to jboss and call them `multihomes -> addresses`, `home -> address`. Let me know what do you think is the best option.